magda-web-client/src/Components/Admin/Connectors.js

```diff
@@ -1,45 +1,143 @@
+import './Connectors.css';
+const uuidV1 = require('uuid/v1');
+type State = {
+  newConnectorName: string,
+  newConnectorType: string,
+  newConnectorFormIsOpen: boolean,
+  error: ?string
+}
 
 class Connectors extends React.Component {
+  state: State = {
+    newConnectorFormIsOpen: false,
+    newConnectorName: '',
+    newConnectorType: '',
+    error: null
+  };
+
+  componentWillMount(){
+    this.props.fetchConnectorsIfNeeded();
+  }
 
   renderByUser(user){
     if(!user){
-      return <Login
-        signInError={
-          this.props.location.state && this.props.location.state.signInError
-        }
-        providers={this.props.providers}
-      />
+      return <div> <Link to="/account">Sign in</Link> as admin to veiw a list of connectors available</div>
     }
     else if(!user.isAdmin){
-      return <div>unauthorised</div>
+      return <div> not authorised </div>
+    }
+    return (<div>
+              <div className='header'><button className='btn btn-primary' onClick={()=>this.openConnectorForm()}>Create a new connector</button></div>
+              <table className='table'><tbody>
+              <tr>
+                <th>Name</th>
+                <th>type</th>
+                <th>url</th>
+                <th>action</th>
+                <th>remove</th>
+              </tr>
+              {this.props.connectors.map(c=>this.renderConnector(c))}</tbody></table>
+              {this.state.newConnectorFormIsOpen && this.renderConnectorForm()}
+            </div>)
+  }
+
+  openConnectorForm(){
+    this.setState({
+      newConnectorFormIsOpen: true
+    })
+  }
+
+  renderConnectorForm(){
+    return  <div className='create-connector-form-wrapper' onClick={()=>{this.setState({newConnectorFormIsOpen: false})}}>
+              <form className='create-connector-form' onClick={(event)=>{event.stopPropagation()}}>
+              {this.state.error && <div className="alert alert-danger">{this.state.error}</div>}
+              <div>
+                <label>
+                    Name:
+                </label>
+                    <input type="text" name="name" onChange={(event)=>{this.setState({newConnectorName: event.target.value, error: null})}} value={this.state.newConnectorName}/>
+              </div>
+              <div>
+                <label>
+                    Type:
+                </label>
+                    <input type="text" name="type" onChange={(event)=>{this.setState({newConnectorType: event.target.value, error: null})}} value={this.state.newConnectorType}/>
+                </div>
+                <input type="button" value="Submit" onClick={()=>this.submitNewConnector()} className='btn btn-primary' />
+            </form>
+          </div>
+  }
+
+  submitNewConnector(){
+    if(this.state.newConnectorName.length > 0 && this.state.newConnectorType.length > 0){
+      this.props.createConnector({
+        name: this.state.newConnectorName,
+        type: this.state.newConnectorType,
+        id: uuidV1()
```

Comment:
Ideally the `id` should be something identifiable that makes sense when you're looking at jobs/pods in kubectl. Maybe even url-encode the name?